Source/WebCore/bindings/js/ScriptController.h

```diff
@@ -108,11 +108,11 @@ class ScriptController : public CanMakeWeakPtr<ScriptController> {
     JSC::JSValue evaluateInWorldIgnoringException(const ScriptSourceCode&, DOMWrapperWorld&);
 
     // This asserts that URL argument is a JavaScript URL.
-    void executeJavaScriptURL(const URL&, RefPtr<SecurityOrigin>, ShouldReplaceDocumentIfJavaScriptURL, bool& didReplaceDocument);
-    void executeJavaScriptURL(const URL& url, RefPtr<SecurityOrigin> securityOrigin = nullptr, ShouldReplaceDocumentIfJavaScriptURL shouldReplaceDocumentIfJavaScriptURL = ReplaceDocumentIfJavaScriptURL)
+    void executeJavaScriptURL(const URL&, RefPtr<SecurityOrigin>&&, ShouldReplaceDocumentIfJavaScriptURL, bool& didReplaceDocument);
```

Comment:
ScriptController.h doesn't include SecurityOrigin.h directly; it gets it by way of Document.h. So when I removed SecurityOrigin.h from Document.h, it caused a build error here.

When passing RefPtr<SecurityOrigin> by value, the copy constructor for RefPtr is called, which will call `ref()` on the SecurityOrigin, which necessitates having a definition for SecurityOrigin rather than a forward declaration. When passing by r-value reference, no methods on SecurityOrigin are called, so a forward declaration is fine.

I could have just #included SecurityOrigin.h here in ScriptController.h, or I could have moved the `executeJavaScriptURL()` method into the implementation file. But this felt like a better solution, since it allows the caller to eliminate a potentially unnecessary refcount churn, and it doesn't require including SecurityOrigin.h.

Reply:
I would have moved the `executeJavaScriptURL` function into the implementation file. It’s really not all that helpful to have it inlined.

And since `executeJavaScriptURL` does not take ownership of the security origin but rather just uses it for a check, changing it to an rvalue reference seems like a step in the wrong direction. If it’s OK for our smart pointer security strategy, I would also consider change the argument to a raw pointer. If not, then the other alternatives above apply.

Reply:
Okay, I'll undo this change and move it into the implementation file.

Reply:
(Also, rather than a raw pointer, this could be two functions, one that takes a `const SecurityOrigin&` and one that doesn't, and it would be the client's responsibility to deal with possibly null values, and not the function's.)